pkg/cas/content_addressable_storage_server_test.go

```diff
@@ -0,0 +1,88 @@
+package cas_test
+
+import (
+	"context"
+	"testing"
+
+	remoteexecution "github.com/bazelbuild/remote-apis/build/bazel/remote/execution/v2"
+	"github.com/buildbarn/bb-storage/internal/mock"
+	"github.com/buildbarn/bb-storage/pkg/blobstore/buffer"
+	"github.com/buildbarn/bb-storage/pkg/cas"
+	"github.com/buildbarn/bb-storage/pkg/digest"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func TestContentAddressableStorageServerBatchReadBlobsSuccess(t *testing.T) {
+	ctrl, ctx := gomock.WithContext(context.Background(), t)
+	defer ctrl.Finish()
+
+	digest1 := digest.MustNewDigest("ubuntu1804", "0000000000000000000000000000000000000000000000000000000000000001", 123)
+	digest2 := digest.MustNewDigest("ubuntu1804", "0000000000000000000000000000000000000000000000000000000000000002", 234)
+
+	request := &remoteexecution.BatchReadBlobsRequest{
+		Digests: []*remoteexecution.Digest{
+			digest1.GetPartialDigest(),
+			digest2.GetPartialDigest(),
+		},
+		InstanceName: "ubuntu1804",
+	}
+
+	contentAddressableStorage := mock.NewMockBlobAccess(ctrl)
+
+	a := make([]byte, 123)
+	buf := buffer.NewValidatedBufferFromByteSlice(a)
+	contentAddressableStorage.EXPECT().Get(ctx, digest1).Return(buf)
+	b := make([]byte, 234)
+	buf2 := buffer.NewValidatedBufferFromByteSlice(b)
+	contentAddressableStorage.EXPECT().Get(ctx, digest2).Return(buf2)
+
+	contentAddressableStorageServer := cas.NewContentAddressableStorageServer(contentAddressableStorage, 1<<16)
+
+	response, err := contentAddressableStorageServer.BatchReadBlobs(ctx, request)
+	if err != nil {
+		t.Errorf("%v", err)
+	}
+	require.Equal(t, &remoteexecution.BatchReadBlobsResponse{
+		Responses: []*remoteexecution.BatchReadBlobsResponse_Response{
+			{
+				Digest: digest1.GetPartialDigest(),
+				Data:   a,
+				Status: status.Convert(nil).Proto(),
+			},
+			{
+				Digest: digest2.GetPartialDigest(),
+				Data:   b,
+				Status: status.Convert(nil).Proto(),
+			},
+		},
```

Comment:
Would it make sense to extend this test to include a third digest for which `ToByteSlice()` returns `NotFound`?

Reply:
Just to clear up my confusion, did you mean where the data buffer is larger than the digest's `SizeBytes` field for example? This would cause `InvalidArgument` from `ToByteSlice`, given that I can't find any way for a `NotFound` to be returned I guess that's what you meant?

Reply:
I'm talking about the case where someone calls BatchReadBlobs(), but passes in one or more digests that are not present in the backing store. Or alternatively: Calling BatchReadBlobs(), even though the backing store is unavailable. That should let the BatchReadBlobs() call succeed, but should cause a non-OK Status to be embedded into the results. You can simulate this by creating a buffer like this:
```go
buf := buffer.NewBufferFromError(codes.NotFound, "The object you requested could not be found")
```